src/atomsbase_interface.jl

```diff
@@ -0,0 +1,73 @@
+#
+# Interface between AtomsBase and GeomOpt that provides 
+# utility functions for manipulating systems.
+#
+# This interface helps defining which particles in the system are considered 
+# optimizable. Note that by default all particles are assumed to be optimizable. 
+# The user can call clamp_atoms to fix atoms whose position should not be optimized.
+#
+# IMPORTANT: Note that we always work in cartesian coordinates.
+#
+export update_positions, update_optimizable_positions, clamp_atoms
+
+
+@doc raw"""
+Creates a new system based on ``system`` but with atoms positions updated 
+to the ones specified in `positions`, using cartesian coordinates.
+"""
+function update_positions(system, positions::AbstractVector{<:AbstractVector{<:Unitful.Length}})
+    particles = [Atom(atom; position) for (atom, position) in zip(system, positions)]
+    AbstractSystem(system; particles)
+end
+
+@doc raw"""
+Creates a new system based on ``system`` with the optimizable coordinates are
+updated to the ones provided (in the order in which they appear in the system),
+cartesian coordinates version..
+"""
+function update_optimizable_positions(system, positions::AbstractVector{<:Unitful.Length})
+    mask = get_optimizable_mask(system)
+    new_positions = deepcopy(position(system))
+    new_positions[mask] = reinterpret(reshape, SVector{3, eltype(positions)},
+                                      reshape(positions, 3, :))
+    update_positions(system, new_positions)
+end
+
+@doc raw"""
+Sets the mask defining which coordinates of the system can be optimized. 
+The mask is a vector of booleans, specifying which of the atoms can be optimized.
+
+By default (when no mask is specified), all particles are assumed optimizable.
+
+"""
```

Comment:
I think this is too restrictive. There are also situations where you want to restrict specific coordinates. For example you can have atom positions be pinned to a surface but be allowed to slide on that surface. Maybe this doesn't have to be provided on the first implementation though. A note or an issue though so it can be done in the future.

Reply:
But the best way to deal with this is to make the mask completely general - e.g. a matrix, and have a simplified `clamp_atoms!` (as below).